lib/legacy.rb

```diff
@@ -17,7 +17,9 @@ module Legacy
     column_names.each do |col|
       if col.in? DATE_FIELDS
         # Convert datetimes to date
-        define_method(col.underscore) { send(col).try(:to_date) }
+        # define_method("#{col.underscore}=") { |val| instance_variable_set("@#{col.underscore}", val) }
+        # define_method(col.underscore) { send(col).try(:to_date) }
+        alias_attribute col.underscore, col
```

Comment:
I intended to clean that up a bit and forgot. Unfortunately, `alias_attribute` was necessary to get the factories working properly, rails attributes work a little weird and I couldn't find a way to dynamically define the assignment method which could reliably actually get the value inserting into the database.

As for the conversion -- because all the dates are passed through the I18n localizer and formatted with the localized format string, it's not really necessary to convert to `Date` objects. They come out at the end-user side the same way, and we aren't using any `Date`-specific methods or anything that prevent us from treating them as `DateTime`'s with a zeroed-out time.